src/QMCDrivers/QMCDriverFactory.cpp

```diff
@@ -294,6 +295,11 @@ std::unique_ptr<QMCDriverInterface> QMCDriverFactory::createQMCDriver(xmlNodePtr
       (das.traces_tag == "none" && (das.new_run_type == QMCRunType::VMC || das.new_run_type == QMCRunType::DMC));
   new_driver->requestTraces(allow_traces);
 
+  //add trace information
+  bool allow_walker_logs = das.walkerlogs_tag == "yes" ||
+      (das.walkerlogs_tag == "none" && (das.new_run_type == QMCRunType::VMC || das.new_run_type == QMCRunType::DMC || das.new_run_type == QMCRunType::VMC_BATCH || das.new_run_type == QMCRunType::DMC_BATCH));
+  new_driver->requestWalkerLogs(allow_walker_logs);
```

Comment:
Complicated control logic. Prefer not to have another switch. I would like to have the following.
if "walkerlog" appears globally, all the drivers honor it. if not, each driver honor the "walkerlog" within the driver input section.

Reply:
Agreeing with Ye here - we use similar logical for maxcpusecs

Reply:
The control logic protects drivers which either don't support walker logging, or shouldn't.

Traces were never enabled for optimizers, for instance.  On another note, I really don't want to be forced to write code to support walker logging in *all* of the legacy drivers.  CSVMC anyone?

Reply:
Fair point

Reply:
Further comments welcome.

Reply:
I was talking about the control logic from the input switch perspective not about implementations. The driver support is expected to be sparse. I thought if a driver doesn't support logging, then certain functions are just no-op.

Still hard to understand the intention of
```
  bool allow_walker_logs = das.walkerlogs_tag == "yes" ||
      (das.walkerlogs_tag == "none" && (das.new_run_type == QMCRunType::VMC || das.new_run_type == QMCRunType::DMC || das.new_run_type == QMCRunType::VMC_BATCH || das.new_run_type == QMCRunType::DMC_BATCH));
```
if I set attribute to "none", It seems that I still get logging when the driver is VMC for example.
If I set it to "yes", then CSVMC will crash? I don't see any "protection".

Reply:
If you want, this could be changed to:

```
bool allow_walker_logs = das.new_run_type == QMCRunType::VMC || das.new_run_type == QMCRunType::DMC || das.new_run_type == QMCRunType::VMC_BATCH || das.new_run_type == QMCRunType::DMC_BATCH);
```

The `das.walkerlogs_tag == "yes"` part just allows one to selectively turn the feature on/off for certain drivers, if desired (user flexibility).

Reply:
Slightly better logic (move the gun away from the user's foot if they intentionally point it there):

```
bool allow_walker_logs = das.walkerlogs_tag == "yes" && (das.new_run_type == QMCRunType::VMC || das.new_run_type == QMCRunType::DMC || das.new_run_type == QMCRunType::VMC_BATCH || das.new_run_type == QMCRunType::DMC_BATCH));
```
With `das.walkerlogs_tag="yes"` being the default.

Reply:
I would like to get rid of `requestWalkerLogs` completely.

Derived class overriding function putWalkerLogs knows what the driver is.
Then its argument also indicates weather logging should be on or off. I don't see the need of additional control.

Reply:
The current walker logging support is limited to VMC/DMC and its functionality and implementation can be completely different for a non-VMC/DMC driver. So its input is more suited for a driver input section rather than globally. If only some VMC/DMC drivers need to turn on logging, the input should be in the driver input section. Right now, we provide convenience to have the input globally and I believe it is better not to another switch to turn it off locally. To me, that is a bad design logic.

Reply:
As for the other protection, setting `allow_walker_logs` correctly to `false` for unsupported drivers is what makes the `WalkerLogManager`/`WalkerLogCollector` operations a no-op.

Reply:
Then default value of `allow_walker_logs` is already `false` to all the unsupported drivers. Why do we even need to set it.

Reply:
Too much details. I would like to discuss the user-facing input logic first and then we can address the implementation issue, maybe at a later PR.

Are you OK with the following input logic
1.  if the global `<walkerlog/>` exists, turn on logging in all the supported drivers.
2.  when the global <walkerlog> doesn't exist, legacy drivers do nothing. batched drivers will look for driver scope `<walkerlog/>` and act on it.